pkg/storage/storage.go

```diff
@@ -620,7 +623,19 @@ func (is *ImageStore) DeleteImageManifest(repo string, reference string) error {
 	outIndex.Manifests = []ispec.Descriptor{}
 
 	for _, m = range index.Manifests {
-		if reference == m.Digest.String() {
+		if isTag {
+			tag, ok := m.Annotations[ispec.AnnotationRefName]
+			if ok && tag == reference {
+				is.log.Debug().Str("deleting tag", tag).Msg("")
+
+				digest = m.Digest
+
+				found = true
+
+				continue
```

Comment:
We really cannot have the same tag repeat in an index.json? or can we?

Reply:
No this loop is appending those manifests which we want to keep so we can't break from here .